### ISSUE TITLE
Substitutions don't work in messages from the main execution loop

### ISSUE THREAD

**Issue by DavidKinder** [DavidKinder/Windows-AGiliTy#5]
When the interpreter's main execution loop calls `gen_sysmsg()` with `context==MSG_MAIN`, substitutions like "<span>$</span>You<span>$</span>" are not being handled, even though they clearly should be.

If the game's data doesn't redefine the message, the message is processed and for "<span>$</span>You<span>$</span>" we'll end up calling `wordcode_match()` with `context==MSG_MAIN`. At the bottom of this function there is code that is clearly meant to handle this case:
```
  if (pronoun_mode && match_str(pvarname,"YOU$")) 
    youme("I","you");
```
but the interpreter never gets that far, as before that there is this line
```
  if (context==MSG_MAIN) return 0;
```
That doesn't look right. The above line is immediately before some substitutions that can't apply to MSG_MAIN messages, like "<span>$</span>NOUN<span>$</span>" and "<span>$</span>VERB<span>$</span>". It looks like the intention of that check is to skip the substitutions that don't apply to MSG_MAIN, but as a result some that do apply are ignored.

The most likely correct fix is to remove the above line, and make sure that the block of code immediately after it doesn't run in the MSG_MAIN case, but the final block that handles pronouns does run.

**Comment by DavidKinder** [comment]
This is the cause of the bug seen in Shades of Gray, where dying produces the message "<span>$</span>You<span>$</span> have died."

**Comment by DavidKinder** [comment]
Fixed by 5bff5c5c52afa19e368999f42ded85cde65f737e.
